src/include/hardware.h

```diff
@@ -73,6 +73,44 @@
 #define WT_WRITE_ONCE(v, val) WT_RELEASE_WRITE_WITH_BARRIER(v, val)
 #endif
 
+/*
+ * The below assembly implements the read-acquire semantic. The if branches get removed at compile
+ * time as the sizeof instruction evaluates at compile time. Additionally ldapr results in a loss of
+ * type checking, to circumvent this we have an if (0) block which contains the direct assignment.
+ * This forces the compiler to type check.
+ *
+ * Depending on the size of the given type we choose the appropriate ldapr variant, additionally the
+ * W register variants are used if possible which map to the lower word of the associated X
+ * register.
+ *
+ * Finally the "Q" constraint is used for the given input operand, this instructs the compiler to
+ * generate offset free ldapr instructions. ldapr instructions, prior to version RCpc 3, don't
+ * support offsets.
+ */
+#ifdef HAVE_RCPC
+#ifndef TSAN_BUILD
+#define WT_ACQUIRE_READ(v, val)                                      \
+    do {                                                             \
+        if (0) {                                                     \
+            v = val;                                                 \
+        }                                                            \
+        if (sizeof((val)) == 1) {                                    \
+            __asm__ volatile("ldaprb %w0, %1" : "=r"(v) : "Q"(val)); \
+        } else if (sizeof((val)) == 2) {                             \
+            __asm__ volatile("ldaprh %w0, %1" : "=r"(v) : "Q"(val)); \
+        } else if (sizeof((val)) == 4) {                             \
+            __asm__ volatile("ldapr %w0, %1" : "=r"(v) : "Q"(val));  \
+        } else if (sizeof((val)) == 8) {                             \
+            __asm__ volatile("ldapr %0, %1" : "=r"(v) : "Q"(val));   \
+        }                                                            \
+    } while (0)
+#else
+#define WT_ACQUIRE_READ(v, val) v = __atomic_load_n(&val, __ATOMIC_ACQUIRE);
```

Comment:
```suggestion
#define WT_ACQUIRE_READ(v, val) v = __atomic_load_n(&(val), __ATOMIC_ACQUIRE);
```

Reply:
Funnily you would also bracket the `v`. And in the `if(0)` branch the `(v) = (val);`

Reply:
da3e8cedf